mixins/harvester-vm/index.js

```diff
@@ -917,6 +971,130 @@ export default {
+
+      for (const row of this.accessCredentials) {
+        let secretRef = row.secretRef;
+
+        if (!secretRef) {
+          secretRef = await this.$store.dispatch('harvester/create', {
+            metadata: {
+              name:            row.secretName,
+              namespace:       vm.metadata.namespace,
+              labels:          { [HCI_ANNOTATIONS.CLOUD_INIT]: 'harvester' },
+              ownerReferences: this.getOwnerReferencesFromVM(vm)
+            },
+            type: SECRET
+          });
+        }
+
+        if (row.source === ACCESS_CREDENTIALS.RESET_PWD) {
+          secretRef.setData(row.username, row.newPassword);
+        }
+
+        if (row.source === ACCESS_CREDENTIALS.INJECT_SSH) {
+          for (const secretId of row.sshkeys) {
+            const keypair = (this.$store.getters['harvester/all'](HCI.SSH) || []).find(s => s.id === secretId);
+
+            secretRef.setData(keypair.metadata.name, keypair.spec.publicKey);
+          }
+        }
+
+        toSave.push(secretRef);
+      }
+
+      try {
+        for (const resource of toSave) {
+          await resource.save();
+        }
+      } catch (e) {
+        new Error(`Function(saveAccessCredentials:save) error ${ e }`);
+
+        return Promise.reject(e);
+      }
+    },
+
+    getAccessCredentialsValidation() {
+      const errors = [];
+
+      for (let i = 0; i < this.accessCredentials.length; i++) {
+        const row = this.accessCredentials[i];
+        const source = row.source;
+
+        if (source === ACCESS_CREDENTIALS.RESET_PWD) {
+          if (!row.username) {
+            const fieldName = this.t('harvester.virtualMachine.input.username');
+            const message = this.t('validation.required', { key: fieldName });
+
+            errors.push(message);
+          }
+
+          if (!row.newPassword) {
+            const fieldName = this.t('harvester.virtualMachine.input.password');
+            const message = this.t('validation.required', { key: fieldName });
+
+            errors.push(message);
+          }
+
+          if (row.newPassword && row.newPassword.length < 6) {
+            const fieldName = this.t('harvester.virtualMachine.input.password');
+            const message = this.t('validation.number.min', { key: fieldName, val: '6' });
+
+            errors.push(message);
+          }
+        } else {
+          if (!row.users || row.users.length === 0) {
+            const fieldName = this.t('harvester.virtualMachine.input.username');
+            const message = this.t('validation.required', { key: fieldName });
+
+            errors.push(message);
+          }
+
+          if (!row.sshkeys || row.sshkeys.length === 0) {
+            const fieldName = this.t('harvester.virtualMachine.input.sshKeyValue');
+            const message = this.t('validation.required', { key: fieldName });
+
+            errors.push(message);
+          }
+        }
+
+        if (errors.length > 0) {
+          break;
+        }
```

Comment:
Can this be deleted?

Reply:
I don't think the loop should be completed, the loop should break when error is found.